persistence/src/test/scala/com/evolutiongaming/akkaeffect/persistence/PersistenceFailureTest.scala

```diff
@@ -0,0 +1,142 @@
+package com.evolutiongaming.akkaeffect.persistence
+
+import akka.actor.Props
+import akka.persistence.journal.AsyncWriteJournal
+import akka.persistence.{AtomicWrite, PersistentRepr}
+import cats.effect.unsafe.implicits.global
+import cats.effect.{IO, Resource}
+import cats.syntax.all.*
+import com.evolutiongaming.akkaeffect.testkit.TestActorSystem
+import com.evolutiongaming.akkaeffect.{Envelope, Receive}
+import com.evolutiongaming.catshelper.LogOf
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalatest.matchers.should.Matchers
+
+import scala.concurrent.Future
+import scala.concurrent.duration.*
+import scala.util.Try
+
+class PersistenceFailureTest extends AnyFunSuite with Matchers {
+
+  sealed trait Event extends FailOnEventJournal.Event
+  object GoodEvent   extends Event { val fail = false }
+  object FailEvent   extends Event { val fail = true  }
+
+  val eventSourced = EventSourcedOf.const {
+    IO {
+
+      val journal  = "fail-on-event-journal"
+      val snapshot = "inmemory-snapshot-store"
+
+      EventSourced(
+        eventSourcedId = EventSourcedId("test"),
+        pluginIds = PluginIds(journal, snapshot),
+        value = RecoveryStarted[Unit] { (_, _) =>
+          Recovering[Unit] {
+            Replay.empty[IO, Event].pure[Resource[IO, *]]
+          } { (_, journaller, _) =>
+            Receive[Envelope[Event]] { envelope =>
+              // persist event in forked thread thus don't fail actor directly
+              journaller.append(Events.of(envelope.msg)).flatten.start.as(false)
+            } {
+              IO(true)
+            }.pure[Resource[IO, *]]
+          }.pure[Resource[IO, *]]
+        }.typeless(
+          sf = _ => IO.unit,
+          ef = e => IO(e.asInstanceOf[Event]),
+          cf = c => IO(c.asInstanceOf[Event]),
+        ).pure[Resource[IO, *]],
+      )
+    }
+  }
+
+  test("EventSourcedActorOf based actor must fail if journal fails to persist message") {
+
+    implicit val log = LogOf.empty[IO]
+
+    TestActorSystem[IO]("testing", none)
+      .use { system =>
+        val persistence = EventSourcedPersistence.fromAkkaPlugins[IO](system, 1.second, 100)
+        def actor       = EventSourcedActorOf.actor[IO, Any, Any](eventSourced, persistence)
+        val ref         = system.actorOf(Props(actor))
+        def tell        = IO(ref ! GoodEvent)
+        def fail        = IO(ref ! FailEvent)
+        def look        = IO.fromFuture(IO(system.actorSelection(ref.path).resolveOne(2.seconds)))
+
+        for {
+          _ <- tell
+          _ <- tell
+          _ <- IO.sleep(1.second)
+          r <- look
+          _  = r shouldBe ref
+          _ <- fail
+          _ <- IO.sleep(1.second)
+          r <- look.attempt
+          _  = r shouldBe a[Left[_, _]]
+        } yield {}
+      }
+      .unsafeRunSync()
```

Comment:
why we do not use something like: https://github.com/typelevel/cats-effect-testing?tab=readme-ov-file#scalatest?

Reply:
because it works without additional magic. Could you share any benefits of one more library?